src/domain/occurrence/__tests__/EditOccurrencePage.test.tsx

```diff
@@ -0,0 +1,256 @@
+import { MockedResponse } from '@apollo/react-testing';
+import { format } from 'date-fns';
+import { advanceTo, clear } from 'jest-date-mock';
+import React from 'react';
+
+import {
+  EventDocument,
+  MyProfileDocument,
+  OccurrenceDocument,
+  OccurrenceNode,
+  PlaceDocument,
+  VenueDocument,
+} from '../../../generated/graphql';
+import * as graphqlFns from '../../../generated/graphql';
+import {
+  fakeEvent,
+  fakeLocalizedObject,
+  fakeLocation,
+  fakeOccurrence,
+  fakeOccurrences,
+  fakeOrganisations,
+  fakePerson,
+  fakePEvent,
+  fakeVenue,
+} from '../../../utils/mockDataUtils';
+import {
+  renderWithRoute,
+  screen,
+  userEvent,
+  waitFor,
+} from '../../../utils/testUtils';
+import { ROUTES } from '../../app/routes/constants';
+import EditOccurrencePage from '../EditOccurrencePage';
+
+const eventName = 'Testitapahtuma';
+const eventMock = fakeEvent({
+  name: fakeLocalizedObject(eventName),
+  startTime: '2020-07-13T05:51:05.761000Z',
+});
+const placeMock = fakeLocation({
+  streetAddress: fakeLocalizedObject('Testikatu'),
+});
+const venueMock = fakeVenue({ hasClothingStorage: true });
+
+advanceTo(new Date(2020, 7, 2));
+
+const occurrence = {
+  amountOfSeats: 30,
+  maxGroupSize: 20,
+  minGroupSize: 10,
+  autoAcceptance: true,
+  startTime: '2020-08-03T09:00:00+00:00',
+  endTime: '2020-08-03T09:30:00+00:00',
+  languages: [
+    { id: 'en', name: 'English', __typename: 'LanguageType' },
+    { id: 'fi', name: 'Finnish', __typename: 'LanguageType' },
+  ] as graphqlFns.LanguageType[],
+};
+
+const fakeOccurrenceOverrides: Partial<OccurrenceNode>[] = [
+  {
+    startTime: '2020-08-03T09:00:00+00:00',
+  },
+  {
+    startTime: '2020-08-04T09:00:00+00:00',
+  },
+  {
+    startTime: '2020-08-05T09:00:00+00:00',
+  },
+  {
+    startTime: '2020-08-06T09:00:00+00:00',
+  },
+  {
+    startTime: '2020-08-07T09:00:00+00:00',
+  },
+];
+
+const apolloMocks: MockedResponse[] = [
+  {
+    request: {
+      query: MyProfileDocument,
+      variables: {},
+    },
+    result: {
+      data: { myProfile: fakePerson({ organisations: fakeOrganisations() }) },
+    },
+  },
+  {
+    request: {
+      query: EventDocument,
+      variables: {
+        id: eventMock.id,
+        include: ['location'],
+      },
+    },
+    result: {
+      data: {
+        event: {
+          ...eventMock,
+          pEvent: fakePEvent({
+            occurrences: fakeOccurrences(5, fakeOccurrenceOverrides),
```

Comment:
Could be little bit more maintainable if you write it like
`occurrences: fakeOccurrences(fakeOccurrenceOverrides.length, fakeOccurrenceOverrides)`